amd/src/replayevents.js

```diff
@@ -0,0 +1,382 @@
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+define(['core/str', 'core/config', 'core/notification', 'core/templates', 'jquery', 'jqueryui'],
+    function(str, mdlcfg, notification, templates, $) {
+
+        /**
+         * Store event ids.
+         */
+        var eventIDs = [];
+
+        /**
+         * Store load HTML snippet.
+         */
+        var loadHTML = '';
+
+        /**
+         * Store replay HTML snippet.
+         */
+        var replayHTML = '';
+
+        /**
+         * Store done HTML snippet.
+         */
+        var doneHTML = '';
+
+        /**
+         * Store failed HTML snippet.
+         */
+        var failedHTML = '';
+
+        /**
+         * Store jquery selectors.
+         */
+        var SELECTORS = {
+            SUBMIT_FORM: '#xapierrorlog_form #id_submitbutton',
+            SELECTS: '#xapierrorlog_form .custom-select',
+            SELECT_ERRORTYPE: '#xapierrorlog_form #id_errortype',
+            SELECT_EVENTNAME: '#xapierrorlog_form #id_eventname',
+            SELECT_RESPONSE: '#xapierrorlog_form #id_response',
+            SELECT_DATAFROM: '#xapierrorlog_form #id_datefrom .custom-select',
+            SELECT_DATATO: '#xapierrorlog_form #id_dateto .custom-select',
+            CHECKBOXES: '#xapierrorlog_form .form-check-input',
+            CHECKBOX_DATEFROM: '#xapierrorlog_form #id_datefrom_enabled',
+            CHECKBOX_DATETO: '#xapierrorlog_form #id_dateto_enabled',
+            RESEND_BUTTON: '#xapierrorlog_form #id_resendselected',
+            REPLAY_EVENTS: '#xapierrorlog_data .reply-event',
+        };
+
+        /**
+         * Added prefix to the replay event id.
+         */
+        var REPLAY_EVENT_ID_PREFIX = 'reply-event-id-';
+
+        var replayevents = {
+
+            /**
+             * Initialisation method called by php js_call_amd()
+             */
+            init: function(ids) {
+                eventIDs = ids;
+
+                this.disableResend();
+                this.updateResend();
+
+                this.addReplyEvents();
+             },
+
+            /**
+             * Register reply an individual event listeners.
+             */
+            addReplyEvents: function() {
+                if($(SELECTORS.REPLAY_EVENTS).length==0) {
+                    return;
+                }
+                this.generateLoadHTML();
+                this.generateDoneHTML();
+                this.generateFailedHTML();
+                this.generateReplayHTML();
+                this.registerReplyEventListeners();
+            },
+
+            /**
+             * Register reply an individual event listeners.
+             */
+            registerReplyEventListeners: function() {
+                var self = this;
+
+                $(SELECTORS.REPLAY_EVENTS).click(function(e) {
+                    e.stopPropagation();
+                    e.preventDefault();
+
+                    self.disableFormControls();
+
```

Comment:
perhaps also I should turn off the pagination (if it is on the page as well) during the request processing.

Reply:
I think the pagination would be necessary (and lower the server load) if a large number of rows were being processed.

Reply:
I think what @lzabo is saying is that the pagination should be disabled along with the rest of the controls when the user selects a single record for replay

Reply:
I was agreeing with @lzabo, @mlynn-lp's comment hadn't appeared for me